bootloader/assets/data/grub.cfg

```diff
@@ -36,14 +39,18 @@ elif [ -n "$kernel_status" ]; then
     save_env kernel_status
 fi
 
-if [ -e $prefix/$kernel ]; then
 menuentry "Run Ubuntu Core 20" {
     # use $prefix because the symlink manipulation at runtime for kernel snap
     # upgrades, etc. should only need the /boot/grub/ directory, not the
     # /EFI/ubuntu/ directory
     chainloader $prefix/$kernel snapd_recovery_mode=run $cmdline_args
 }
-else
-    # nothing to boot :-/
-    echo "missing kernel at $prefix/$kernel!"
-fi
+menuentry "Fallback on failed update" {
```

Comment:
maybe this should have `--id failed-kernel-load-fallback-entry` and then earlier `set fallback=failed-kernel-load-fallback-entry`. I feel that identifying entries by numbers can become tricky in the long run.

Reply:
I agree, and I actually tried with the id first, but for some reason that didn't work and I had to use the entry number. I've checked and apparently focal's grub should support it nonetheless, so I would give it a another try, maybe I did something wrong.

Reply:
FTR, I tried again with --id but no luck. Looking at the code, it seems that this is not really possible, at least for focal (it seems that `grub_strtoul()` gets called to get the value for the "fallback" variable):

https://git.launchpad.net/ubuntu/+source/grub2/tree/grub-core/normal/menu.c?h=applied/ubuntu/focal-updates#n337
https://git.launchpad.net/ubuntu/+source/grub2/tree/grub-core/normal/menu.c?h=applied/ubuntu/focal-updates#n171

Reply:
Yeah, unfortunately it's still the same in master: https://github.com/rhboot/grub2/blob/971dd6599d838ed21f38f7261c7bdda59f9bbdae/grub-core/normal/menu.c#L336